test/controllers/api/v2/smart_proxies_controller_test.rb

```diff
@@ -250,4 +250,30 @@ class Api::V2::SmartProxiesControllerTest < ActionController::TestCase
     show_response = ActiveSupport::JSON.decode(@response.body)
     assert_match(/Exception message/, show_response['error']['message'])
   end
+
+  context 'import subnets' do
+    setup do
+      SmartProxy.expects(:find).with('foo').returns(mock('proxy'))
+    end
+
+    test 'import subnets failed' do
+      Subnet::Ipv4.expects(:import).returns(nil)
+      post :import_subnets, params: { :id => 'foo' }, session: set_session_user
+      assert_response :unprocessable_entity
+      show_response = ActiveSupport::JSON.decode(@response.body)
+      assert_match 'Proxy does not have a DHCP feature', show_response['error']['message']
+    end
+
+    test 'render no_content if subnets were not found' do
+      Subnet::Ipv4.expects(:import).returns([])
+      post :import_subnets, params: { :id => 'foo' }, session: set_session_user
+      assert_response :no_content
+    end
+
+    test 'import subnets succeeded' do
```

Comment:
Could we also check if we really get an array with subnets and their attributes, such as `.id`, `name`.